test/test.js

```diff
@@ -164,6 +164,44 @@ function behavesLikeParseObjectOnBeforeDelete(typeName, ParseObjectOrUserSubclas
   });
 }
 
+function behavesLikeParseObjectOnAfterSave(typeName, ParseObjectOrUserSubclass) {
+  context('when object has afterSave hook registered', () => {
+    const errorMessage = 'Error in afterSave hook test';
+    let didAfterSave = false;
+    function afterSavePromise(request) {
+      if (request.object.get('error')) {
+        return Parse.Promise.error(errorMessage);
+      }
+      didAfterSave = true;
+      return Parse.Promise.as();
+    }
+
+    it('runs the hook after saving the model and persists the object', () => {
+      ParseMockDB.registerHook(typeName, 'afterSave', afterSavePromise);
+
+      const object = new ParseObjectOrUserSubclass();
+      assert(!object.has('cool'));
```

Comment:
Good catch - thanks 👍 
To make sure it's always false at the beginning of a case, I've added it into the beforeEach function.